File: 151001/Shlyk/SpringCalculator/spring-calculator-rest/src/main/java/by/bsuir/poit/dc/rest/api/dto/mappers/UserMapper.java
package by.bsuir.poit.dc.rest.api.dto.mappers;

import by.bsuir.poit.dc.rest.api.dto.request.UpdateUserDto;
import by.bsuir.poit.dc.rest.api.dto.response.UserDto;
import by.bsuir.poit.dc.rest.model.User;
import org.mapstruct.*;

import java.util.List;

/**
 * @author Paval Shlyk
 * @since 31/01/2024
 */
@Mapper(unmappedTargetPolicy = ReportingPolicy.ERROR,
    componentModel = MappingConstants.ComponentModel.SPRING,
    nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
public abstract class UserMapper {
    @Mapping(target = "id", ignore = true)
    @Mapping(target = "news", ignore = true)
    public abstract User toEntity(UpdateUserDto dto);

    @Mapping(target = "id", ignore = true)
    @Mapping(target = "news", ignore = true)
    public abstract User partialUpdate(
	@MappingTarget User user,
	UpdateUserDto dto);

    @Named("toDto")
    public abstract UserDto toDto(User user);

    @IterableMapping(qualifiedByName = "toDto")
    public abstract List<UserDto> toDtoList(List<User> users);
}

File: 151003/Barylko/Kafka/Module.Publisher/Mappers/UserMapper.cs
using Publisher.Dto.Request.CreateTo;
using Publisher.Dto.Response;
using Publisher.Models;
using Riok.Mapperly.Abstractions;
namespace Publisher.Mappers;

[Mapper]
public static partial class UserMapper
{
    public static partial User ToEntity(this CreateUserRequestTo createUserRequestTo);
    public static partial UserResponseTo ToResponse(this User User);
    public static partial IEnumerable<UserResponseTo> ToResponse(this IEnumerable<User> Users);
}